types/mydecimal_test.go

```diff
@@ -778,3 +887,44 @@ func (s *testMyDecimalSuite) TestMaxOrMin(c *C) {
 		c.Assert(dec.String(), Equals, tt.result)
 	}
 }
+
+func benchmarkMyDecimalToBinOrHashKey(b *testing.B, toHashKey bool) {
+	cases := []string{
+		"1.000000000000", "3", "12.000000000", "120",
+		"120000", "100000000000.00000", "0.000000001200000000",
+		"98765.4321", "-123.456000000000000000",
+		"0", "0000000000", "0.00000000000",
+	}
+	decs := make([]*MyDecimal, 0, len(cases))
+	for _, ca := range cases {
+		var dec MyDecimal
+		if err := dec.FromString([]byte(ca)); err != nil {
+			b.Fatal(err)
+		}
+		decs = append(decs, &dec)
+	}
+
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		for _, dec := range decs {
+			var err error
+			if toHashKey {
+				_, err = dec.ToHashKey()
+			} else {
+				prec, frac := dec.PrecisionAndFrac()
+				_, err = dec.ToBin(prec, frac)
+			}
+			if err != nil {
+				b.Fatal(err)
+			}
+		}
+	}
+}
+
+func BenchmarkMyDecimalToBin(b *testing.B) {
+	benchmarkMyDecimalToBinOrHashKey(b, false)
```

Comment:
For preciseness, it's better to write benchmark separately, because the `if` branch can also consumes some CPU cycles.

Reply:
solved